server/src/main/java/org/opensearch/action/admin/cluster/node/stats/NodeStats.java

```diff
@@ -142,6 +142,9 @@ public class NodeStats extends BaseNodeResponse implements ToXContentFragment {
     @Nullable
     private SearchPipelineStats searchPipelineStats;
 
+    @Nullable
+    private Boolean isNodeUnderDuress;
```

Comment:
> Shall we also include a metric suggesting whether the node is under duress or not? It will be useful for investigation. IsNodeUnderDuress can be added in the SearchBackpressureStats pojo.

@PritLadani Do you have a strong justification to have this exposed as part of an API? Logs are there for debuggability, and I feel that boolean value doesn't convey much information for users to act upon. Please note that this is a one-way door decision, and unless we have a strong justification today, it's better to avoid gratuitous changes.

@Bukhtawar What do you think?

Reply:
Had a discussion with @ketanv3 offline and we decided not to include this metric pre-maturely now. We can have some framework around node health and expose this and other similar metrics there.
@kaushalmahi12 Let's remove it for now.

Reply:
Came across this PR (https://github.com/opensearch-project/OpenSearch/pull/9890) which adds node performance trackers and exposes its metrics in the node stats API. Please check with the authors if they're already planning to have something similar to this.